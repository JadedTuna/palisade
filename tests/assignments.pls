in {
    high secret: bool;
    high num: int;
}
out {
    high x: bool;
    low y: int;
}

x = true;
x = secret;
x = true || secret;

y = num;
y = 42;
y = 42 + num;
