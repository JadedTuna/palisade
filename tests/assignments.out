[1;34mnote: [0mlabel of [1;34mx[0m set to [1;93mlow[0m
  10 | x [1;34m=[0m true;
       [1;34m~~[0m[1;34m^[0m
[1;34mnote: [0mlabel of [1;34mx[0m set to [1;93mhigh[0m
  11 | x [1;34m=[0m secret;
       [1;34m~~[0m[1;34m^[0m
[1;34mnote: [0mlabel of [1;34my[0m set to [1;93mhigh[0m
  14 | y [1;34m=[0m num;
       [1;34m~~[0m[1;34m^[0m
[1;34mnote: [0mlabel of [1;34my[0m set to [1;93mlow[0m
  15 | y [1;34m=[0m 42;
       [1;34m~~[0m[1;34m^[0m
[1;34mnote: [0mlabel of [1;34my[0m set to [1;93mhigh[0m
  16 | y [1;34m=[0m 42 + num;
       [1;34m~~[0m[1;34m^[0m
[[1;32m OK [0m] [1;93mhigh[0m [1;34mx[0m is [1;93mhigh[0m
[[1;31mFAIL[0m] [1;93mlow [0m [1;34my[0m is [1;93mhigh[0m
